src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const arrOfStates = [{ ...state }];
+
+  for (const { type, extraData, keysToRemove } of actions) {
+    const stateCopy = { ...arrOfStates[arrOfStates.length - 1] };
```

Comment:
If there leave empty array, can you delete arrOfStates.shift() on function bottom?

```suggestion
  const arrOfStates = [];

  for (const { type, extraData, keysToRemove } of actions) {
    const currentState =  arrOfStates.length ? { ...arrOfStates[arrOfStates.length - 1] } : { ... state };
```